xija/limits.py

```diff
@@ -0,0 +1,60 @@
+def get_limit_spec(limit):
+    """
+    Based on an input *limit* string of the form:
+
+    <system>.<type>.<direction>.<qualifier>
+
+    (where the "qualifier" element is optional) return the various pieces 
+    into a dictionary for convenience.
+    """
+    words = limit.split(".")
+    if len(words) < 3:
+        raise RuntimeError(f"{limit} is not a valid limit string!")
+    limit_spec = {
+        "system": words[0],
+        "type": words[1],
+        "direction": words[2]
+    }
+    if len(words) == 4:
+        limit_spec["qualifier"] = words[3]
+    else:
+        limit_spec["qualifier"] = None
+    return limit_spec
+
+
+def get_limit_color(limit):
+    """
+    Based on an input *limit* string of the form:
+    
+    <system>.<type>.<direction>.<qualifier>    
+    
+    (where the "qualifier" element is optional) return the color to
+    be used for plotting.
+    """
+    limit_spec = get_limit_spec(limit)
+    if limit_spec["qualifier"] is not None:
```

Comment:
I think this logic assumes anything with a qualifier is ACIS. The logic here is bound to get messy, so I think it might be easier to just enumerate the colors in an ordered dict and use wildcards. Something like:
```
LIMIT_COLORS = {
'*.acisi': 'blue',  # could be more specific if that makes sense
...   # and so on
'odb.caution.*': 'gold',
'safety.caution.*': 'gold',
'odb.warning.*': 'red',
...
'planning.caution.*': 'dodgerblue',
...
'*': 'black'   # ?? Either have a catchall or else issue a warning (but not exception)
}
```
This way it is easy to expand, make special cases, and so on. The first match would win.

Reply:
I implemented something more or less identical to this. See new version.